log.go

```diff
@@ -4,207 +4,102 @@ import (
 	"errors"
 	"fmt"
 	"io"
-	"os"
-	"runtime"
-	"strings"
-	"sync/atomic"
 )
 
-var pid = os.Getpid()
-var currentSeverity Severity
+var loggers []Logger
 
-// Severity implementation is borrowed from glog, uses sync/atomic int32
-type Severity int32
-
-const (
-	SeverityInfo Severity = iota
-	SeverityWarn
-	SeverityError
-	SeverityFatal
-)
-
-var severityName = map[Severity]string{
-	SeverityInfo:  "INFO",
-	SeverityWarn:  "WARN",
-	SeverityError: "ERROR",
-	SeverityFatal: "FATAL",
-}
-
-// get returns the value of the severity.
-func (s *Severity) Get() Severity {
-	return Severity(atomic.LoadInt32((*int32)(s)))
-}
-
-// set sets the value of the severity.
-func (s *Severity) Set(val Severity) {
-	atomic.StoreInt32((*int32)(s), int32(val))
-}
-
-// less returns if this severity is greater than passed severity
-func (s *Severity) Gt(val Severity) bool {
-	return s.Get() > val
-}
-
-func (s Severity) String() string {
-	n, ok := severityName[s]
-	if !ok {
-		return "UNKNOWN SEVERITY"
-	}
-	return n
-}
-
-func SeverityFromString(s string) (Severity, error) {
-	s = strings.ToUpper(s)
-	for k, val := range severityName {
-		if val == s {
-			return k, nil
-		}
-	}
-	return -1, fmt.Errorf("unsupported severity: %s", s)
-}
-
-// Logger is a unified interface for all loggers.
+// Logger is an interface that should be implemented by all loggers wishing to participate
+// in the logger chain initialized by this package.
 type Logger interface {
-	Infof(format string, args ...interface{})
-	Warningf(format string, args ...interface{})
-	Errorf(format string, args ...interface{})
-	Fatalf(format string, args ...interface{})
-
+	Infof(string, ...interface{})
+	Warningf(string, ...interface{})
+	Errorf(string, ...interface{})
```

Comment:
I propose deleting `Infof`, `Warningf`, and `Errorf` from the interface. We don't use them. We use `Writer` and `FormatMessage` instead.